### PR TITLE
feat(release): add Linux arm64 OCI target

### LINKED ISSUE #1346: feat(release): add authenticated Linux arm64 OCI target
## Problem

Astrid needs a native Linux arm64 OCI target built from the same authenticated
release artifacts as its CLI installer. The amd64 image must not be emulated or
silently reused for arm64, and rebuilding a moving source checkout inside the
image would create a second release path.

## Scope

- Add a native Linux arm64 OCI image definition that installs verified bytes
  from an exact immutable Astrid GitHub release.
- Verify signed release metadata and the arm64 artifact digest before
  installation.
- Run and test on a native arm64 runner.
- Match the non-root, read-only-compatible, capability-dropped security
  contract established by the amd64 target.
- Require a signed distribution to be mounted or preseeded by the operator.
- Produce arm64 SBOM, vulnerability scan, provenance, and image-signature
  inputs in CI.

The amd64 image and canonical multi-architecture manifest are separate work.

## Acceptance criteria

- Native arm64 CI proves the image contains the authenticated arm64 release
  bytes and starts the daemon in the foreground with logs on stderr.
- Startup fails closed for missing or invalid signed distribution/release
  verification.
- The image runs rootless without privileged mode or a Docker socket.
- No mutable `latest`, `stable`, `dev`, or `nightly` image tag is published by
  this change.

### PR DESCRIPTION
**Pull request by joshuajbouw**
## Linked Issue

Closes #1346

## Summary

Adds a native Linux arm64 OCI target assembled from authenticated, immutable Astrid release bytes. The image and CI preserve the amd64 rootless and fail-closed runtime contract without emulation, mutable tags, or registry publication.

## Changes

- Add a native `ubuntu-24.04-arm` OCI workflow with explicit host and Docker architecture assertions and no QEMU/binfmt path.
- Authenticate the exact arm64 release archive and signed metadata before building.
- Export one exact OCI archive, strictly bind its sole arm64 manifest and blobs, and reuse the digest-qualified image for runtime tests, Trivy, and Syft.
- Add rootless/read-only/capability-dropped ARM container tests, including signed-distro, entrypoint, and restricted-runtime negative cases.
- Produce checksum-bound OCI, SBOM, scan, release, and image-binding evidence; keep optional signing behind protected manual `main` dispatch.
- Extend shared acquisition and container-contract tests while preserving the amd64 defaults and test-image alias behavior.
- Document the arm64 target and add an `[Unreleased]` changelog entry.

## Verification

- 73 focused Python tests pass across release acquisition, OCI binding, amd64 container contracts, and arm64 container contracts.
- Fresh live arm64 release authentication passed both Sigstore checks.
- `actionlint` passes.
- `shellcheck` passes.
- `bash -n` and Python compilation pass.
- Full-SHA action-pin policy and `git diff --check` pass.
- Independently reviewed for native execution, artifact identity, evidence binding, rootless security, and non-publication scope.

Native arm64 image execution in the dedicated GitHub Actions job is the merge gate.

## Checklist

- [x] Linked to an issue
- [x] CHANGELOG.md updated (entry under `[Unreleased]` — or `[Unreleased]` rolled into a version section for a release PR; not applicable to docs/CI-only changes)

### CODE REVIEW

File: container/arm64/test.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+export ASTRID_OCI_TEST_PLATFORM=linux/arm64
+export ASTRID_OCI_TEST_ARCHITECTURE=arm64
+export ASTRID_OCI_TEST_LABEL=arm64
+
+exec container/amd64/test.sh "$@"
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply container/arm64/test.sh]: Fixed in 03a39e23. The ARM wrapper now derives an absolute repository root from `BASH_SOURCE[0]` and execs the shared harness through that path. The ARM contract test covers location-independent invocation, and shellcheck plus all 73 focused tests pass.

File: .github/workflows/oci-arm64.yml [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+          [[ "$VERSION" =~ ^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)$ ]]
+          [[ "$SOURCE_COMMIT" =~ ^[0-9a-f]{40}$ ]]
+          {
+            echo "VERSION=$VERSION"
+            echo "SOURCE_COMMIT=$SOURCE_COMMIT"
+            echo "IMAGE=astrid-runtime:${VERSION}-arm64"
+          } >> "$GITHUB_ENV"
+
+      - name: Install pinned BLAKE3 verifier
+        run: cargo install b3sum --version 1.8.5 --locked
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply .github/workflows/oci-arm64.yml:91]: Fixed in 03a39e23. The pinned Rust 1.95 toolchain setup now runs before `cargo install b3sum`, so the native runner no longer relies on incidental preinstalled Cargo state; the same toolchain remains available for the later WASM fixture build.
